lib/enigma.rb

```diff
@@ -0,0 +1,63 @@
+class Enigma
+  attr_reader :key,
+              :date
+
+  def initialize
+    @key = nil
+    @date = nil
+  end
+
+  def key_object(key)
+    if key.nil?
+      random_key = rand(00000..99999).to_s.rjust(5, "0")
+      @key = Key.new(random_key)
+    elsif key.is_a?(Key)
+      @key = key
+    else
+      @key = Key.new(key)
+    end
+  end
+
+  def date_object(date)
```

Comment:
Should this be a responsibility of enigma? Should DateCode be responsible fo instantiating itself and making the decision what to do with the information given to it?